dotnet/SqlClient
# Upgrade / align packages in test projects

- Upgrade packages only used in test projects to latest versions
- Upgrade XUnit / MSTest SDK
- Remove unused variables
  - MicrosoftNETCoreRuntimeCoreCLRVersion
  - MicrosoftDotNetPlatformAbstractionsVersion
  - SystemTextEncodingCodePagesVersion
- Remove reference to xunit.runner.utility, this should not be linked as per https://github.com/xunit/visualstudio.xunit/issues/409#issuecomment-2480747672

According to https://github.com/dotnet/SqlClient/pull/2379#issuecomment-1979544650, the Remote Executor package and XUnit was held back by .NET6, now that .NET6 is removed, this should work fine
